src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const clone = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(clone, action.extraData);
+      result.push(Object.assign({}, clone));
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach(property => {
+        delete clone[property];
+      });
+      result.push(Object.assign({}, clone));
+    }
+
+    if (action.type === 'clear') {
+      Object.keys(clone).forEach(key => {
+        delete clone[key];
+      });
+      result.push(Object.assign({}, clone));
+    }
+  }
```

Comment:
do not repeat your self
```suggestion
  for (const action of actions) {
    if (action.type === 'addProperties') {
      Object.assign(clone, action.extraData);
    }

    if (action.type === 'removeProperties') {
      action.keysToRemove.forEach(property => {
        delete clone[property];
      });
    }

    if (action.type === 'clear') {
      Object.keys(clone).forEach(key => {
        delete clone[key];
      });
    }
    result.push({ ...clone });
  }
```